test/report/test-report-config.js

```diff
@@ -0,0 +1,65 @@
+// Flags: --experimental-report --diagnostic-report-on-fatalerror --diagnostic-report-on-signal --diagnostic-report-uncaught-exception
+'use strict';
+const common = require('../common');
+common.skipIfReportDisabled();
+const assert = require('assert');
+
+common.expectWarning('ExperimentalWarning',
+                     'report is an experimental feature. This feature could ' +
+                     'change at any time');
+
+// Verify that process.report.directory behaves properly.
+assert.strictEqual(process.report.directory, '');
+process.report.directory = __dirname;
+assert.strictEqual(process.report.directory, __dirname);
+process.report.directory = {};
+assert.strictEqual(process.report.directory, '[object Object]');
```

Comment:
? This is kind of weird. Should we instead prevent `report.directory` being set to anything other than a string?

Reply:
I don't have a strong opinion personally. I just thought that coercion seemed like a more JavaScripty user experience with setters/getters.

Reply:
I'd prefer the stricter string-only requirement.